### PR TITLE
feat:added history feature

### PR DESCRIPTION
**Pull request by shrushti2000**
Implemented history feature
1. When user views a video , it will be added to history
2. Users can view history 
3. Users can remove videos from history
4. Users can clear all history
ezgif com-gif-maker (39)

### CODE REVIEW

**Review by nyyshaaa — APPROVED** [review]
Code and UI looks clean. Nicely done!

File: src/App.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { Route, Routes } from "react-router-dom";
 import { Footer, Header } from "./components/index";
```
Comment (bharati-21) [inline src/App.js]: You can omit `/index`. It's optional 👍🏽

File: src/App.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { Route, Routes } from "react-router-dom";
...
 import { Homepage, Signin, Signup, VideoListingpage,VideoPage,Playlistpage, SinglePlaylistPage, WatchLater, LikedVideosPage } from "./pages/index.js";
```
Comment (bharati-21) [inline src/App.js]: Suggest using a formatter such a prettier.

File: src/App.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { Route, Routes } from "react-router-dom";
...
 import "./App.css";
+import { HistoryPage } from "./pages/HistoryPage/HistoryPage";
```
Comment (bharati-21) [inline src/App.js:15]: You can add this to the index.js file within pages/ if you forgot to do it 👍🏽

File: src/Context/Context.js [hunk 4]
```diff
@@ -46,7 +47,7 @@ const Context = ({ children }) => {
         fetchData()
     }, [])
     return (
-        <StateContext.Provider value={{ categories: state.categories, videos: state.videos, showtoast: state.showtoast, sortByLatest: state.sortByLatest, selectedCategory: state.selectedCategory,playlists:state.playlists,showPlaylistModal:state.showPlaylistModal,currentVideo:currentVideo,setCurrentVideo:setCurrentVideo,watchlater:state.watchlater,likedVideos:state.likedVideos,dispatch }}>{children}</StateContext.Provider>
+        <StateContext.Provider value={{ categories: state.categories, videos: state.videos, showtoast: state.showtoast, sortByLatest: state.sortByLatest, selectedCategory: state.selectedCategory,playlists:state.playlists,showPlaylistModal:state.showPlaylistModal,currentVideo:currentVideo,setCurrentVideo:setCurrentVideo,watchlater:state.watchlater,likedVideos:state.likedVideos,watchedVideos:state.watchedVideos,dispatch }}>{children}</StateContext.Provider>
```
Comment (bharati-21) [inline src/Context/Context.js]: You could have used spread on the state here.

File: src/Utility/historyService.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import axios from "axios"
+
+export const addVideoToHistory = async (token, video, dispatch, navigate) => {
+    console.log('added')
+    console.log(token)
+    if (token) {
```
Comment (bharati-21) [inline src/Utility/historyService.js]: Please remove logs.

File: src/Utility/historyService.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            const res = await axios.post('/api/user/history',
+                { video: video },
```
Comment (bharati-21) [inline src/Utility/historyService.js]: You can use object shorthand here.

File: src/components/HistoryVideoCard/HistoryVideoCard.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    const openVideoPage = (e) => {
+        e.preventDefault();
+        if (e.target.classList.value !== 'svg-inline--fa fa-ellipsis-vertical cta-icon') {
+            navigate(`/video/${videoItem._id}`)
+        }
+    }
+
+    return (
+        <>
+            <div className='video-item-container flex-vt' >
```
Comment (bharati-21) [inline src/components/HistoryVideoCard/HistoryVideoCard.js]: A fragment is used when you have multiple sibling elements and you need to wrap them around a parent but don't need an extra element.

Here, your div with video-item-container is already a parent here. I don't see the purpose of using a fragment here.
Any reason?

File: src/components/VideoCard/VideoCard.js [hunk 8]
```diff
@@ -57,13 +58,14 @@ const VideoCard = ({ videoItem }) => {
...
-                                </div></>}
+                                </div></>: <><div className='remove-from-watchlater flex-vt' onClick={() => removeFromlikedVideos(token, dispatch, videoItem._id, showCTAcontainer, setShowCTAcontainer,navigate)}>
+                                    <div className='flex-hz'><FontAwesomeIcon icon={faTrash}></FontAwesomeIcon><p >Remove from Liked Videos</p></div>   
+                                </div></> }
                             </> : <> <div className='video-cta-container flex-vt'>
                                 <div className='video-item-cta-item flex-hz'><FontAwesomeIcon className='video-item-cta-icon' icon={faClock}></FontAwesomeIcon><p className='video-item-cta-text' onClick={() => addToWatchLater(token, dispatch, videoItem, showCTAcontainer, setShowCTAcontainer,navigate)}>Add to Watch Later</p></div>
                                 <div className='video-item-cta-item flex-hz' onClick={displayPlaylistModal}><FontAwesomeIcon className='video-item-cta-icon' icon={faCirclePlay}></FontAwesomeIcon><p className='video-item-cta-text'>Add to Playlist</p></div>
```
Comment (bharati-21) [inline src/components/VideoCard/VideoCard.js]: Please use a formatter here

File: src/pages/HistoryPage/HistoryPage.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import { Link, Navigate, useNavigate } from 'react-router-dom'
+import { Sidebar } from '../../components'
+import { useAuth } from '../../Context/AuthProvider'
+import { useData } from '../../Context/Context'
+import { HistoryVideoCard } from '../../components/HistoryVideoCard/HistoryVideoCard'
```
Comment (bharati-21) [inline src/pages/HistoryPage/HistoryPage.js]: You could add this component withing your component common file as well.

**Review by bharati-21 — COMMENTED** [review]
I have a dropped a few suggestions.

Your code is self explanatory! 
My main suggestion would be to use a formatter as it's difficult to scroll horizontally.
Otherwise all looks good.

**Comment by shrushti2000** [comment]
Thanks for the PR review @nisha-lamboria @bharati-21 
Will surely work upon the suggestions and format the code with prettier as well.

**Pull request merged**